operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java

```diff
@@ -100,19 +100,24 @@ public static Condition buildRebalanceCondition(String type) {
         if (resource.getMetadata().getGeneration() != null)    {
             status.setObservedGeneration(resource.getMetadata().getGeneration());
         }
+        // condition contains a timestamp so using a set does not help
         List<Condition> conditionList = new ArrayList<>();
         Condition condition = StatusUtils.buildCondition(type, conditionStatus, null);
         conditionList.add(condition);
 
-        if (status.getConditions() != null) {
-            status.getConditions().forEach(cond -> {
-                if ("UnknownFields".equals(cond.getReason()) || "DeprecatedFields".equals(cond.getReason())) {
+        status.getConditions().forEach(cond -> {
+            if ("UnknownFields".equals(cond.getReason()) || "DeprecatedFields".equals(cond.getReason())) {
+                if (!conditionAlreadyPresent(conditionList, cond)) {
+                    cond.setLastTransitionTime(iso8601Now());
                     conditionList.add(cond);
                 }
-            });
-        }
+            }
```

Comment:
Could you please explain what this code does? It is common to have the messages with the same reason present multiple times. For example:

```yaml
    - lastTransitionTime: "2021-02-15T13:05:24.039453Z"
      message: In API version kafka.strimzi.io/v1beta1 the metrics property at path
        spec.kafka.metrics has been deprecated, and should now be configured using
        spec.kafka.metricsConfig.
      reason: DeprecatedFields
      status: "True"
      type: Warning
    - lastTransitionTime: "2021-02-15T13:05:24.039453Z"
      message: In API version kafka.strimzi.io/v1beta1 the metrics property at path
        spec.zookeeper.metrics has been deprecated, and should now be configured using
        spec.zookeeper.metricsConfig.
      reason: DeprecatedFields
      status: "True"
      type: Warning
```

So I'm not entirely sure if what you are doing here is right.

Reply:
https://gist.github.com/im-konge/b26e068fce4c5c1395717ef7c4a807ab

Reply:
That is without of context => so hard to understand the issue there. But I do not think it goes against my comment => I'm not arguing that you should have there the same warning 100 times. I'm saying that the Reason is not expected to be unique in general and should not be treated as such.

Yesterday when I checked this code in IDE it seemed to be used by KafkaRebalance only ... so maybe it never worked properly 🤷🏻‍♂️.